add docs on migrating block documents (#8085)
PrefectHQ/prefect: docs/concepts/blocks.md

In the above example, the values for `AWSCredentials` are saved with `my_s3_bucket` and will not be usable with any other blocks.

### Handling updates to custom `Block` types

> Let's say that you now want to add a field `bucket_folder` to your custom `S3Bucket` block that represents the default path to read and write objects from (this fields exists on [our implementation](https://github.com/PrefectHQ/prefect-aws/blob/main/prefect_aws/s3.py#L292)).

Reviewer: ```suggestion
Let's say that you now want to add a `bucket_folder` field to your custom `S3Bucket` block that represents the default path to read and write objects from (this field exists on [our implementation](https://github.com/PrefectHQ/prefect-aws/blob/main/prefect_aws/s3.py#L292)).
```

We can add the new field to the class definition:

```python hl_lines="4"
class S3Bucket(Block):
    bucket_name: str
    credentials: AWSCredentials
    bucket_folder: str = None
    ...
```

> ...and [register the updated block type with your Prefect Orion server](#registering-blocks-for-use-in-the-prefect-ui) (whether cloud or self-hosted).

Reviewer: ```suggestion

Then [register the updated block type](#registering-blocks-for-use-in-the-prefect-ui) with either Prefect Cloud or your self-hosted Prefect Orion server.
```
